### PR TITLE
Post Details

### LINKED ISSUE #43: View Post Details
As a reader, I would like to see the content of a Post so I can read it.

**Given** a user is viewing a list of Posts  
**When** they select a post to read  
**Then** they should be directed to a Post Detail page that shows the Post Details.

Post Details include:

* Title
* Header image (if exists)
* Content
* Publication date (MM/DD/YYYY)
* Author's Display Name

### PR DESCRIPTION
**Pull request by cjgochanour**
#### Changes Made
1. Added file `PostDetails.js` to `client/src/components` directory.
2. Modified file `postManager.js` to include `getPostById` that calls `PostController.cs`'s Get{id} method
3. Added `GetById(int id)` method to `PostRepository`
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout cg-postDetails
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
4. Test app functionality.
    > Log in to app if not logged in already
    > Navigate to post list
    > Click the title of a post to be navigated to its detail view
    > As per #43 you will see the image header, post title, post author, post contents, and post's publish date
5. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

Closes #43

### CODE REVIEW

**Review by user1hunter — APPROVED** [review]
Works as intended and I see no errors in the code. Looks good to me.

File: Tabloid/Repositories/PostRepository.cs [hunk 1]
```diff
@@ -53,6 +53,58 @@ public List<Post> GetApprovedPosts()
...
+
+                    Post post = null;
+                    if (reader.Read())
+                    {
+                        post = NewPostFromReader(reader);
+                        while (reader.Read())
+                        {
+                            post.Tags.Add(new Tag
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("TagId")),
```
Comment (joshdbarton) [inline Tabloid/Repositories/PostRepository.cs]: DbUtils?

File: Tabloid/Controllers/PostController.cs [hunk 2]
```diff
@@ -27,9 +27,14 @@ public IActionResult Get()
 
         // GET api/<PostController>/5
         [HttpGet("{id}")]
-        public string Get(int id)
+        public IActionResult Get(int id)
```
Comment (joshdbarton) [inline Tabloid/Controllers/PostController.cs:33]: Should any authed user be able to see any post? If not we might need to figure out the logic to make sure that this route is protected. For example, if a post isn't published who should be able to see it?

**Review by joshdbarton — CHANGES_REQUESTED** [review]
see comments

File: Tabloid/Repositories/PostRepository.cs (hunk 1 above)
Comment (cjgochanour) [reply Tabloid/Repositories/PostRepository.cs]: updated

File: Tabloid/Controllers/PostController.cs (hunk 2 above)
Comment (cjgochanour) [reply Tabloid/Controllers/PostController.cs:33]: This has been changed. If a post is not yet published or is not yet approved, only admins and the post's author may view the details of the post. All users logged in may view details of approved, already published posts.

**Review by joshdbarton — APPROVED** [review]
:shipit:

**Pull request merged**